packages/smart-contracts/helpers/proxyFactory.ts

```diff
@@ -0,0 +1,54 @@
+import { ethers } from "ethers";
+
+import dedgeCompoundManagerDef from "../artifacts/DedgeCompoundManager.json";
+
+import { Address } from "./types";
+
+const IDedgeCompoundManager = new ethers.utils.Interface(
+  dedgeCompoundManagerDef.abi
+);
+
+const buildAndEnterMarkets = (
+  dacProxyFactory: ethers.Contract,
+  dedgeCompoundManager: Address,
+  cTokensToEnter: Address[],
+  gasLimit: number = 4000000
+): Promise<any> => {
+  const enterMarketsCallbackData = IDedgeCompoundManager.functions.enterMarkets.encode(
+    [cTokensToEnter]
+  );
+
+  return dacProxyFactory["buildAndEnterMarkets(address,bytes)"](
+    dedgeCompoundManager,
+    enterMarketsCallbackData,
+    {
+      gasLimit
+    }
+  );
+};
+
+const buildAndEnterMarketsOwner = (
+  owner: Address,
+  dacProxyFactory: ethers.Contract,
+  dedgeCompoundManager: Address,
+  cTokensToEnter: Address[],
+  gasLimit: number = 4000000
+): Promise<any> => {
+  const enterMarketsCallbackData = IDedgeCompoundManager.functions.enterMarkets.encode(
+    [cTokensToEnter]
+  );
+
+  return dacProxyFactory["buildAndEnterMarkets(address,address,bytes)"](
+    owner,
+    dedgeCompoundManager,
+    enterMarketsCallbackData,
+    {
+      gasLimit
+    }
+  );
+};
+
+export default {
+  buildAndEnterMarkets,
+  buildAndEnterMarketsOwner
```

Comment:
What's the difference between these two?

Reply:
One creates it for msg.sender, one creates it for `owner`